File: Assets/Standard Assets/Scripts/CubeTest.cs
using UnityEngine;
using System.Collections;

public class CubeTest : MonoBehaviour {

	int cont = 0;
	Vector2 posx;
	Vector2 posy;

	void Start () {
	
		posx = new Vector2(transform.position.x+1, transform.position.y);
		posy = new Vector2(transform.position.x, transform.position.y+1);
	}
	
	void Update () {


		transform.position = posx;

	
	}
}

File: Assets/Scripts/CubeTest.js

var contx = 0;
var conty= 0;

function Start () {

}

function Update () {

	contx +=1;
	
	if (contx >= 5){
		transform.position.y +=1 * Time.deltaTime;
		contx = 0;
		conty +=1;
	}
	
	if (conty < 5) {
		transform.position.x -=1 * Time.deltaTime;
		conty = 1;
	} 
	else {
		transform.position.x +=1 * Time.deltaTime;
	}

}